storey/writers.py

```diff
@@ -670,6 +670,7 @@ async def _do(self, event):
             await self._table._persist(_PersistJob(event.key, data_to_persist, self._handle_completed, event))
         else:
             data_to_persist = self._event_to_writer_entry(event)
-            self._table._update_static_attrs(event.key, data_to_persist)
+            async with self._table._get_lock(event.key):
```

Comment:
better move the lock inside the function no? so we won't miss calls to update_static_attrs
like in _FunctionWithStateFlow

Reply:
No can do, because it gets called in `_lazy_load_key_with_aggregates`, and trying to obtain the lock again will deadlock. But I can add locking in `_FunctionWithStateFlow`.